File: client/src/app/sm/model/field.model.ts
import { RuleModel } from './rule.model';

export class FieldModel {
  constructor() {
    this.field = '';
    this.title = '';
    this.updateType = 'hide';
    this.view = true;
    this.insert = true;
    this.visible = true;
    this.dataType = 'verchar';
    this.fieldType = 'text';
    this.maxlength = 0;

    // 索引
    this.sort = 0;

    // 排列
    this.align = 'center';
    this.halign = 'center';

    // 宽度
    // this.width = '';

    // 排序
    this.order = '';

    // 不可重复
    this.duplicated = false;

    // 格式化
    this.dataFormat = '';
    this.radio = false;
    this.checkbox = false;
    this.pattern = '';
  }

  // 字段
  field: string;
  halign: string;
  falign: string;
  valign: string;
  sortable: boolean;
  cardVisible: boolean;
  switchable: boolean;
  clickToSelect: boolean;
  formatter: string;
  sortName: string;
  searchable: boolean;

  // 名称
  title: string;

  // 修改选项（隐藏,可用,不可用）
  updateType: string;

  // 可查看
  view: boolean;

  // 可添加
  insert: boolean;

  // 列表可见
  visible: boolean;
  checkbox: boolean;
  radio: boolean;

  // 数据类型
  dataType: string;

  // 字段类型
  fieldType: string;

  // 最大长度
  maxlength: number;

  // 索引
  sort: number;

  // 排列
  align: string;

  // 宽度
  width: string;

  // 排序
  order: string;

  // 不可重复
  duplicated: boolean;

  // 格式化
  dataFormat: string;
  titleTooltip: string;
  pattern: string;
  rowspan: number;
  colspan: number;
  searchFormatter: boolean;
  escape: boolean;
}

File: cms-sm/src/main/java/com/sophia/cms/sm/model/FieldModel.java
package com.sophia.cms.sm.model;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import org.hibernate.validator.constraints.NotBlank;

import java.io.Serializable;

/**
 * Created by lenovo on 2017/8/13.
 */
@Data
public class FieldModel implements Serializable {

    @NotBlank(message = "字段不能为空值")
    private String field;

    @NotBlank(message = "名称不能为空值")
    private String title;

    @NotBlank(message = "名称不能为空值")
    private String updateType;
    private boolean view;
    private boolean insert;
    private boolean visible;
    private String dataType;
    private String fieldType;
    private int maxlength;
    private int sort;
    private boolean radio;
    private boolean checkbox;

    @ApiModelProperty(value = "提醒")
    private String titleTooltip;
    private int rowspan;
    private int colspan;
    private String align;
    private String halign;
    private String falign;
    private String valign;
    private String width;
    private boolean sortable;
    private String order;
    private boolean cardVisible;
    private boolean switchable;

    private boolean duplicated;
    private boolean clickToSelect;
    private String formatter;
    private String sortName;
    private boolean searchable;
    private boolean searchFormatter;
    private boolean escape;
    private String dataFormat;

    @ApiModelProperty(value = "表单校验器")
    private String pattern;

    @ApiModelProperty(value = "脱敏类型:1=中文名，2=身份证，3=手机号，4=电子邮件，5=银行卡")
    private Integer masking;

    @JsonSerialize(using = ToStringSerializer.class)
    @ApiModelProperty(value = "数据字典")
    private String dict;

//    private String footerFormatter;
//    private String events;
//    private String sorter;
//    private String cellStyle;
}
